### PR TITLE
Update DG for Group

### PR DESCRIPTION
**Pull request by djleong01**
Add DG details for group and grouplist.

Also include changes to transaction section based on feedback.

### CODE REVIEW

File: docs/diagrams/Group.png
Comment (1simjustin) [inline docs/diagrams/Group.png:1]: Add plantuml code in a sep file for version control purposes

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -219,18 +219,141 @@ storage.saveAllData();
...
 <ins>Class Structure</ins>
 
+The `Group` class has the following attributes:
+* *memberList*: A MemberList object representing the list of Members in the group.
+* *transactionList*: A TransactionList object representing the list of Transactions in the group.
+* *storage*: A StorageHandler object representing the storage handler for the group.
+* *groupName*: A string representing the name of the group.
+* *transactionSolution*: An array list collection of Subtransaction objects representing the least transactions solution to solving all debts in the group.
+
+The `GroupList` class has the following attribute:
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:234]: Attribute plural

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -219,18 +219,141 @@ storage.saveAllData();
...
+
+The detailed class diagram for `Group` and `GroupList` can be found below.
+
+![Group Class Diagram](diagrams/Group.png)
+
 <ins>Constructor</ins>
 
+The Group constructor creates a group object with the given group name and initializes a new member list, transaction list, storage handler. The latter is used to ensure that data across groups are kept discrete.
+
+Key arguments of the Group constructor are a string `groupName`.
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:249]: Plurality again

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -219,18 +219,141 @@ storage.saveAllData();
...
+import longah.node.GroupList;
+
+// Creating a new GroupList instance
+GroupList groupList = new GroupList();
+
+// Creating a new group and adding it to the group list
+groupList.createGroup();
+
+// Adding more groups to the list
+groupList.createGroup();
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:325]: Now that I look at this it looks rather... interesting. Would suggest either removing these 3 lines or adding a method to `GroupList` with String as its param and using that here

Comment (1simjustin) [reply docs/DeveloperGuide.md:325]: Considering `createGroup()` doesnt even execute if the grouplist is empty this just seems erroneous actually

Comment (djleong01) [reply docs/DeveloperGuide.md:325]: changing to `addGroup` in next commit

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -215,18 +215,143 @@ storage.saveAllData();
...
+The `GroupList` class has the following attributes:
+* *activeGroup*: A Group object representing the currently active group.
+* *groupList*: An array list collection of Group objects representing the list of groups stored in the application.
+
+
+<ins>Implementation Details</ins>
+
+The detailed class diagram for `Group` and `GroupList` can be found below.
+
+![Group Class Diagram](diagrams/Group Class.png)
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:239]: Image is broken
image

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -215,18 +215,143 @@ storage.saveAllData();
...
 <ins>Class Structure</ins>
 
+The `Group` class has the following attributes:
+* *memberList*: A MemberList object representing the list of Members in the group.
+* *transactionList*: A TransactionList object representing the list of Transactions in the group.
+* *storage*: A StorageHandler object representing the storage handler for the group.
+* *groupName*: A string representing the name of the group.
+* *transactionSolution*: An array list collection of Subtransaction objects representing the least transactions solution to solving all debts in the group.
+
+The `GroupList` class has the following attributes:
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:230]: Standardisation: end with fullstop instead of colon

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -323,130 +448,88 @@ members.delete("Bob");
...
-* Called whenever a new transaction is added to the transaction list.
+* *lender*: A member object representing the lender in the transaction.
+* *transactionTime*: A DateTime object representing the time of the transaction. (optional)
+* *subtransactions*: An ArrayList of Subtransaction objects, representing individual borrowings within the transaction.
 
-`Transaction(Member lender, ArrayList<Subtransaction> subtransactions,
-MemberList members)`
-* Constructs a transaction instance using specified lender, subtransactions, member list.
-* This constructor is used for storage methods only.
+The TransactionList class has the following attributes.
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:474]: Plurality

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -323,130 +448,88 @@ members.delete("Bob");
...
+* *transactions*: An ArrayList of Transaction objects representing the list of transactions in a group.
 
-<ins>Transaction Methods</ins>
+<ins>Implementation Details</ins>
 
-- *parseTransaction*: Parses the user input to extract lender and borrowers, then adds them to the transaction.
+The detailed class diagram for `Transaction` and `TransactionList` can be found below.
 
-- *addBorrower*: Adds a borrower to the transaction.
+![Transaction Class Diagram](diagrams/Transaction Class.png)
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:482]: Broken image again
image

Comment (1simjustin) [reply docs/DeveloperGuide.md:482]: dont leave space betw file name

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -323,130 +448,88 @@ members.delete("Bob");
...
-
-- *addTransaction*: Parses user input and adds a new transaction to the list.
-
-- *getTransactionListSize*: Returns the size of the transaction list.
-
-- *remove*: Removes a transaction from the list based on the index.
+The TransactionList class has the following key methods.
 
+- *addTransaction*: Adds a new transaction to the list based on user input.
+- remove: Removes a transaction from the list based on the index.
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:504]: Check formatting, some arent italacised

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -323,130 +448,88 @@ members.delete("Bob");
...
 
-- *toStorageString*: Converts the transaction to a string format for storage.
+<ins>Methods</ins>
 
-- *getSubtransactions*: Returns the list of subtransactions in the transaction.
+The Transaction class has the following key methods.
 
+- *parseTransaction*: Parses the user input to extract lender and borrowers, then adds them to the transaction.
+- *addBorrower*: Adds a borrower to the transaction.
+- *toStorageString*: Converts the transaction to a string format for storage.
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:497]: tostoragestring non-key
addborrower is helper
both can be removed

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -323,130 +448,88 @@ members.delete("Bob");
...
 - *findDebts*: Finds all debts owed by a specified member.
-
 - *deleteMember*: Deletes a member from all transactions in the list.
 
 <ins>Usage Example</ins>
 
-Adding a new transaction:
-![addTransaction.png](diagrams%2FaddTransaction.png)
+The diagram below illustrates a sample usage scenario of adding a transaction: 
+![addTransaction.png](diagrams/addTransaction.png)
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:520]: Diagram needs to include all the return arrows

**Comment by 1simjustin** [comment]
I didnt notice ytd but dont include logging in sequence diagram.

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -471,15 +472,15 @@ The Transaction class has the following attributes.
 * *transactionTime*: A DateTime object representing the time of the transaction. (optional)
 * *subtransactions*: An ArrayList of Subtransaction objects, representing individual borrowings within the transaction.
 
-The TransactionList class has the following attributes.
+The TransactionList class has the following attribute.
```
Comment (1simjustin) [inline docs/DeveloperGuide.md:475]: code segment for TransactionList

**Review by 1simjustin — APPROVED** [review]
LGTM

**Pull request merged**